### PR TITLE
work for the last week

### PR DESCRIPTION
**Pull request by mertid**

### CODE REVIEW

File: DigitalVelocity/DigitalVelocity/MenuOptions.swift [hunk 1]
```diff
@@ -110,17 +110,18 @@ private var _menuOptions : [MenuOption] = [MenuOption]()
  */
 class MenuOption{
     
-    
     var title : String!
     var storyboardId : String!
+    var accessibilityId : String!
```
Comment (jalakoo) [inline DigitalVelocity/DigitalVelocity/MenuOptions.swift]: Did we still need the accessibilityId in this class?

**Comment by mertid** [comment]
updated

File: DigitalVelocity/DigitalVelocity/shared/controller/SignIn_VC.swift [hunk 2]
```diff
@@ -12,8 +12,32 @@ public let loginSuccessfulNotification = "loginSuccessful"
...
-    @IBOutlet weak var emailTextField:UITextField!
+    @IBOutlet weak var emailTextField:UITextField! {
+        didSet{
+            let bottomLine = CALayer()
+            bottomLine.frame = CGRectMake(0.0, emailTextField.frame.height - 1, emailTextField.frame.width, 1.0)
+            bottomLine.backgroundColor = UIColor.whiteColor().CGColor
+            emailTextField.borderStyle = UITextBorderStyle.None
+            emailTextField.layer.addSublayer(bottomLine)
+            
+            let str = NSAttributedString(string: "(Account)", attributes: [NSForegroundColorAttributeName:UIColor.lightGrayColor()])
```
Comment (jalakoo) [inline DigitalVelocity/DigitalVelocity/shared/controller/SignIn_VC.swift]: Change this to "(Registration Email)"

File: DigitalVelocity/DigitalVelocity/shared/controller/Settings_VC.swift [hunk 3]
```diff
@@ -10,7 +10,21 @@ import UIKit
...
 
-    @IBOutlet weak var emailTextField: UITextField!
+    @IBOutlet weak var emailTextField: UITextField!{
+        didSet{
+            let bottomLine = CALayer()
+            bottomLine.frame = CGRectMake(0.0, emailTextField.frame.height - 1, emailTextField.frame.width, 1.0)
+            bottomLine.backgroundColor = UIColor.whiteColor().CGColor
+            emailTextField.borderStyle = UITextBorderStyle.None
+            emailTextField.layer.addSublayer(bottomLine)
+            let str = NSAttributedString(string: "(registration email)", attributes: [NSForegroundColorAttributeName:UIColor.lightGrayColor()])
```
Comment (jalakoo) [inline DigitalVelocity/DigitalVelocity/shared/controller/Settings_VC.swift]: Use true caps here, like the rest of the placeholders in the app.

File: DigitalVelocity/DigitalVelocity/shared/controller/Demo_VC.swift [hunk 4]
```diff
@@ -30,8 +87,27 @@ class Demo_VC: UIViewController, UITextFieldDelegate {
         environmentTextField.delegate = self
         audienceStreamTextField.delegate = self
         
+        accountTextField.becomeFirstResponder()
```
Comment (jalakoo) [inline DigitalVelocity/DigitalVelocity/shared/controller/Demo_VC.swift]: Any particular reason accountTextField becomes first responder here vs. in viewDidAppear or viewDidLoad?

File: DigitalVelocity/DigitalVelocity/shared/controller/Demo_VC.swift [hunk 5]
```diff
@@ -10,10 +10,67 @@ import UIKit
 
 class Demo_VC: UIViewController, UITextFieldDelegate {
```
Comment (jalakoo) [inline DigitalVelocity/DigitalVelocity/shared/controller/Demo_VC.swift]: This class fairly small, but get into the habit of breaking up logic blocks of code with something like:

// MARK: LIFECYCLE
// MARK: PUBLIC INSTANCE
// MARK: PRIVATE INSTANCE
// MARK: HELPERS

We will add which types of section headers to use in our next update to the styleguides.

File: DigitalVelocity/DigitalVelocity/shared/controller/Demo_VC.swift [hunk 6]
```diff
@@ -30,8 +87,27 @@ class Demo_VC: UIViewController, UITextFieldDelegate {
         environmentTextField.delegate = self
         audienceStreamTextField.delegate = self
         
+        accountTextField.becomeFirstResponder()
+        
         setupMenuNavigationForController()
         self.prepopulateTextFields()
+
```
Comment (jalakoo) [inline DigitalVelocity/DigitalVelocity/shared/controller/Demo_VC.swift]: Nice touch here.

File: DigitalVelocity/DigitalVelocity/shared/controller/Demo_VC.swift [hunk 7]
```diff
@@ -10,10 +10,67 @@ import UIKit
 
 class Demo_VC: UIViewController, UITextFieldDelegate {
     
-    @IBOutlet weak var accountTextField: UITextField!
-    @IBOutlet weak var profileTextField: UITextField!
-    @IBOutlet weak var environmentTextField: UITextField!
-    @IBOutlet weak var audienceStreamTextField: UITextField!
+    @IBOutlet weak var backgroundImageView: UIImageView!
+    @IBOutlet weak var accountTextField: UITextField!{
+        didSet{
```
Comment (jalakoo) [inline DigitalVelocity/DigitalVelocity/shared/controller/Demo_VC.swift]: This text field is the only one in my test that re-inserted the placeholder text after the clear button was tapped.

File: DigitalVelocity/DigitalVelocity/shared/controller/Demo_VC.swift [hunk 8]
```diff
@@ -76,22 +153,63 @@ class Demo_VC: UIViewController, UITextFieldDelegate {
...
         Analytics.trackEvent("Demo Save Button Tapped")
         
-        print(accountEntryTemp)
+        print(environmentEntryTemp)
        
     }
     
+    func textFieldDidBeginEditing(textField: UITextField) {
+       
+        // better ui handling textfield
```
Comment (jalakoo) [inline DigitalVelocity/DigitalVelocity/shared/controller/Demo_VC.swift]: This block probably not necessary, if removed, corrects the strange behavior where the the placeholders are not re-inserted for certain textfields.

File: DigitalVelocity/DigitalVelocity/shared/controller/Demo_VC.swift [hunk 9]
```diff
@@ -101,4 +219,32 @@ class Demo_VC: UIViewController, UITextFieldDelegate {
...
+        let alertController = UIAlertController(title: "Unable to Update Your Account", message: "You must fill in all fields in order to save", preferredStyle: .Alert)
+        let okAction = UIAlertAction(title: "Ok", style: .Default, handler: nil)
+        
+        alertController.addAction(okAction)
+        
+        self.presentViewController(alertController, animated: true, completion: nil)
+    }
+
+    func presentSuccessAlert(){
+        let alertController = UIAlertController(title: "Success", message: "Your account has been updated", preferredStyle: .Alert)
```
Comment (jalakoo) [inline DigitalVelocity/DigitalVelocity/shared/controller/Demo_VC.swift]: Update message to "Demo account updated"

File: DigitalVelocity/DigitalVelocity/shared/controller/Demo_VC.swift (hunk 5 above)
Comment (jalakoo) [inline DigitalVelocity/DigitalVelocity/shared/controller/Demo_VC.swift]: UIKeyboard should dismiss if menu button tapped. Pre-existing issue with Settings_VC as well.

File: DigitalVelocity/DigitalVelocity/shared/controller/Settings_VC.swift [hunk 10]
```diff
@@ -10,7 +10,21 @@ import UIKit
 
 class Settings_VC: UIViewController {
```
Comment (jalakoo) [inline DigitalVelocity/DigitalVelocity/shared/controller/Settings_VC.swift]: See if you get the UIKeyboard to dismiss when the user taps the Menu Button (hamburger) and when they swipe left to bring up the side menu.

File: DigitalVelocity/DigitalVelocityUITests/DigitalVelocityUITests.swift [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+        // In UI tests it’s important to set the initial state - such as interface orientation - required for your tests before they run. The setUp method is a good place to do this.
+    }
+    
+    override func tearDown() {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+        super.tearDown()
+    }
+    
+
+    func testButtons(){
```
Comment (jalakoo) [inline DigitalVelocity/DigitalVelocityUITests/DigitalVelocityUITests.swift]: UI Events to add to test:

- Agenda:  Tapping on a cell, star an event, return to Agenda
- Agenda:  Filter by star events, unfilter by star events (this actually results in a crash we need to fix)
- Agenda: Tap on a cells map button -> (tap on all button options) -> return to agenda
- Event Location: Tap on all available buttons before proceeding (moving and pinching map view would be a plus)
- Sponsors: Tap on  cell -> return to sponsors
- (later, do not add for this pull request) Sponsors: we'll want to add a tap on the demo button ->  enter an email -> ok

File: DigitalVelocity/DigitalVelocity/shared/controller/Settings_VC.swift [hunk 12]
```diff
@@ -10,7 +10,21 @@ import UIKit
...
 
-    @IBOutlet weak var emailTextField: UITextField!
+    @IBOutlet weak var emailTextField: UITextField!{
```
Comment (jalakoo) [inline DigitalVelocity/DigitalVelocity/shared/controller/Settings_VC.swift]: Email text field is missing the field underline that is preset in the demo and the login views.

File: DigitalVelocity/DigitalVelocity/shared/controller/Demo_VC.swift (hunk 7 above)
Comment (jalakoo) [inline DigitalVelocity/DigitalVelocity/shared/controller/Demo_VC.swift]: FIXED

File: DigitalVelocity/DigitalVelocity/shared/controller/NavigationController.swift [hunk 13]
```diff
@@ -44,6 +44,7 @@ class NavigationController: UINavigationController {
             
             let font = FontAwesomeHelper.fontAwesomeForSize(28)
             let attributes  = NSDictionary(object: font, forKey: NSFontAttributeName)
+            self.menuButton().accessibilityIdentifier = "Menu Button"
```
Comment (jalakoo) [inline DigitalVelocity/DigitalVelocity/shared/controller/NavigationController.swift]: This should be _menuButton?.accessibilityLabel = "Menu Button"

File: DigitalVelocity/DigitalVelocityUITests/DigitalVelocityUITests.swift [hunk 14]
```diff
@@ -0,0 +1,127 @@
...
+        let app = XCUIApplication.init()
+ 
+        app.buttons["Menu Button"].tap()
```
Comment (jalakoo) [inline DigitalVelocity/DigitalVelocityUITests/DigitalVelocityUITests.swift]: Current build: UI test on my system always fails on this line.

File: DigitalVelocity/DigitalVelocity/shared/controller/Settings_VC.swift (hunk 10 above)
Comment (jalakoo) [inline DigitalVelocity/DigitalVelocity/shared/controller/Settings_VC.swift]: Presuming we couldn't get this working.

Comment (mertid) [inline DigitalVelocity/DigitalVelocity/shared/controller/Settings_VC.swift]: No there is the issue with the sliding menu not being in the navigation stack

**Pull request merged**